### PR TITLE
Move ProxyShuffleInternalManagerBase to api [databricks]

### PR DESCRIPTION
**Pull request by gerashegalov**
This PR is taken from #9444 to separate the aspect of moving ProxyShuffleInternalManagerBase
to the sql-plugin-api module

- moves ProxyShuffleInternalManagerBase to sql-plugin-api module along
      with the required dependencies including ShimLoader
- Adds dbdeps profile to sql-plugin-api because we now have to depend 
  on spark classes in the API module as well
- Leave the parts of ShimLoader that cannot be moved now as ShimLoaderTemp
    
Signed-off-by:  @gerashegalov

### CODE REVIEW

**Comment by gerashegalov** [comment]
build

**Comment by gerashegalov** [comment]
build

**Review by jlowe — APPROVED** [review]

**Comment by jlowe** [comment]
build

File: sql-plugin-api/src/main/scala/com/nvidia/spark/rapids/ShimLoader.scala [hunk 1]
```diff
@@ -149,17 +135,14 @@ object ShimLoader extends Logging {
         // fast path
         logInfo(s"findURLClassLoader found a URLClassLoader $urlCl")
         Option(urlCl)
-      case replCl if replCl.getClass.getName == "org.apache.spark.repl.ExecutorClassLoader" ||
-          replCl.getClass.getName == "org.apache.spark.executor.ExecutorClassLoader" =>
-        // Spark 3.5.0 changed the package of ExecutorClassLoader so we check for it being
-        // either old package name or new one.
+      case replCl if replCl.getClass.getName == "org.apache.spark.repl.ExecutorClassLoader" =>
```
Comment (abellina) [inline sql-plugin-api/src/main/scala/com/nvidia/spark/rapids/ShimLoader.scala:138]: is this change on purpose?

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/ShimLoaderTemp.scala [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+import com.nvidia.spark.GpuCachedBatchSerializer
+import com.nvidia.spark.rapids.delta.DeltaProbe
+import com.nvidia.spark.rapids.iceberg.IcebergProvider
+
+import org.apache.spark.internal.Logging
+import org.apache.spark.sql.rapids.{AdaptiveSparkPlanHelperShim, ExecutionPlanCaptureCallbackBase}
+
+// TODO contains signatures that cannot yet be moved to sql-plugin-api
+object ShimLoaderTemp extends Logging {
```
Comment (abellina) [inline sql-plugin/src/main/scala/com/nvidia/spark/rapids/ShimLoaderTemp.scala:27]: > // TODO contains signatures that cannot yet be moved to sql-plugin-api
is there a plan to move them, and then remove the `Temp` side of it? I guess I am unclear on how apis are supposed to move to sql-plugin-api

File: sql-plugin-api/src/main/scala/com/nvidia/spark/rapids/ShimLoader.scala (hunk 1 above)
Comment (gerashegalov) [reply sql-plugin-api/src/main/scala/com/nvidia/spark/rapids/ShimLoader.scala:138]: this is a merge conflict resolution error. thanks for the catch!

Should be caught by #9504 when it goes in

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/ShimLoaderTemp.scala (hunk 2 above)
Comment (gerashegalov) [reply sql-plugin/src/main/scala/com/nvidia/spark/rapids/ShimLoaderTemp.scala:27]: Yes, everything here accessed via the code before the plugin is initialized, and should move to sql-plugin-api
after refactoring. Currently there are still a lot of intertwined jdep dependencies with the shimmed area.
If a signature is inherently unshimmable we need to do more invasive refactoring.

**Review by abellina — DISMISSED** [review]
LGTM

**Review by jlowe — APPROVED** [review]

**Comment by gerashegalov** [comment]
had to resolve a conflict again

**Comment by gerashegalov** [comment]
build

**Review by jlowe — APPROVED** [review]
